src/main/scala/scala/compat/java8/converterImpl/MakesSteppers.scala

```diff
@@ -5,27 +5,117 @@ import language.implicitConversions
+    case StepperShape.DoubleValue => new Stepper.UnboxingDoubleStepper(stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Double]])
+    case StepperShape.ByteValue   => new Stepper.UnboxingByteStepper  (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Byte  ]])
+    case StepperShape.ShortValue  => new Stepper.UnboxingShortStepper (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Short ]])
+    case StepperShape.CharValue   => new Stepper.UnboxingCharStepper  (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Char  ]])
+    case StepperShape.FloatValue  => new Stepper.UnboxingFloatStepper (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Float ]])
+    case _                        => throw new NotImplementedError("AnyStepper must be handled in `stepper` implementations")
+  }).asInstanceOf[S]
 }
 
-trait MakesValueStepper[+T <: Stepper[_]] extends Any {
-  /** Generates a fresh stepper of type `T` over map values */
-  def valueStepper: T
+/** Superclass for `MakesKeyalueStepper` implementations which support parallelization. At least the `AnyStepper` case must be
+  * implemented, all others default to building an `AnyStepper` and putting an unboxing conversion on top. */
+trait MakesKeyValueParStepper[K, V] extends Any with MakesKeyValueStepper[K, V, EfficientSubstep] {
+  def keyStepper[S <: Stepper[_]](implicit ss: StepperShape[K, S]) = (ss match {
+    case StepperShape.IntValue    => new Stepper.UnboxingIntStepper   (keyStepper(StepperShape.anyStepperShape[K]).asInstanceOf[AnyStepper[Int   ]]) with EfficientSubstep
+    case StepperShape.LongValue   => new Stepper.UnboxingLongStepper  (keyStepper(StepperShape.anyStepperShape[K]).asInstanceOf[AnyStepper[Long  ]]) with EfficientSubstep
+    case StepperShape.DoubleValue => new Stepper.UnboxingDoubleStepper(keyStepper(StepperShape.anyStepperShape[K]).asInstanceOf[AnyStepper[Double]]) with EfficientSubstep
+    case StepperShape.ByteValue   => new Stepper.UnboxingByteStepper  (keyStepper(StepperShape.anyStepperShape[K]).asInstanceOf[AnyStepper[Byte  ]]) with EfficientSubstep
+    case StepperShape.ShortValue  => new Stepper.UnboxingShortStepper (keyStepper(StepperShape.anyStepperShape[K]).asInstanceOf[AnyStepper[Short ]]) with EfficientSubstep
+    case StepperShape.CharValue   => new Stepper.UnboxingCharStepper  (keyStepper(StepperShape.anyStepperShape[K]).asInstanceOf[AnyStepper[Char  ]]) with EfficientSubstep
+    case StepperShape.FloatValue  => new Stepper.UnboxingFloatStepper (keyStepper(StepperShape.anyStepperShape[K]).asInstanceOf[AnyStepper[Float ]]) with EfficientSubstep
+    case _                        => throw new NotImplementedError("AnyStepper case must be handled in `keyStepper` implementations")
+  }).asInstanceOf[S with EfficientSubstep]
+
+  def valueStepper[S <: Stepper[_]](implicit ss: StepperShape[V, S]) = (ss match {
```

Comment:
I think this whole thing could be just `ss.unboxing(valueStepper(StepperShape.anyStepperShape[V]))`